src/de.rs

```diff
@@ -8,6 +8,10 @@ use crate::int_key::IntKey;
 pub trait KeyDeserialize {
     type Output: Sized;
 
+    /// The number of key elements is used for the deserialization of compound keys.
+    /// It should be equal to PrimaryKey::key().len()
+    const KEY_ELEMS: u16;
```

Comment:
This is a breaking change since the trait is public

Reply:
Yes it is. Unfortunately there is no solutions to this bug that does not introduce breaking changes. If you look at one of my earlier suggestions I mentioned that creating a new trait called `KeyElems`, which contains the const, and then using that as a trait bound for `KeyDeserialize for (T, U)` would make this change _less_ braking for most users. I still think that's a good solution.

Reply:
Thank you! We'll check next week how we can ensure the 1.x release series remains free of breaking changes. It might take a while until this can be released. So I recommend using the main branch from GitHub as a dependency to unblock you.

Reply:
Will do. If you go with the new trait solution, then the changes will only be braking specifically to users who are directly exposed to the bug. Isn't it a good thing that they will now receive a compiler error, forcing them to address the issue? That is definitely the route I would take.

Reply:
Yeah it might be the best option. I need to look into it with a fresh head again. For now I just want to make sure this does not break happy users when released.